src/Symfony/Component/HttpKernel/Tests/DataCollector/ConfigDataCollectorTest.php

```diff
@@ -58,6 +65,53 @@ public function testLegacy()
         $this->assertSame('name', $c->getApplicationName());
         $this->assertNull($c->getApplicationVersion());
     }
+
+    public function testCollectWithoutKernel()
+    {
+        $c = new ConfigDataCollector();
+        $c->collect(new Request(), new Response());
+
+        $this->assertSame('n/a', $c->getEnv());
+        $this->assertSame('n/a', $c->isDebug());
+        $this->assertSame('config', $c->getName());
+        $this->assertMatchesRegularExpression('~^'.preg_quote($c->getPhpVersion(), '~').'~', \PHP_VERSION);
+        $this->assertMatchesRegularExpression('~'.preg_quote((string) $c->getPhpVersionExtra(), '~').'$~', \PHP_VERSION);
+        $this->assertSame(\PHP_INT_SIZE * 8, $c->getPhpArchitecture());
+        $this->assertSame(class_exists(\Locale::class, false) && \Locale::getDefault() ? \Locale::getDefault() : 'n/a', $c->getPhpIntlLocale());
+        $this->assertSame(date_default_timezone_get(), $c->getPhpTimezone());
+        $this->assertSame(Kernel::VERSION, $c->getSymfonyVersion());
+        $this->assertSame(4 === Kernel::MINOR_VERSION, $c->isSymfonyLts());
+        $this->assertNull($c->getToken());
+        $this->assertSame(\extension_loaded('xdebug'), $c->hasXDebug());
+        $this->assertSame(\extension_loaded('Zend OPcache') && filter_var(ini_get('opcache.enable'), \FILTER_VALIDATE_BOOLEAN), $c->hasZendOpcache());
+        $this->assertSame(\extension_loaded('apcu') && filter_var(ini_get('apc.enabled'), \FILTER_VALIDATE_BOOLEAN), $c->hasApcu());
+        $this->assertSame(sprintf('%s.%s', Kernel::MAJOR_VERSION, Kernel::MINOR_VERSION), $c->getSymfonyMinorVersion());
+        $this->assertSame($this->determineSymfonyState(), $c->getSymfonyState());
+
+        $eom = \DateTime::createFromFormat('d/m/Y', '01/'.Kernel::END_OF_MAINTENANCE)->format('F Y');
+        $eol = \DateTime::createFromFormat('d/m/Y', '01/'.Kernel::END_OF_LIFE)->format('F Y');
+        $this->assertSame($eom, $c->getSymfonyEom());
+        $this->assertSame($eol, $c->getSymfonyEol());
+    }
+
+    private function determineSymfonyState(): string
```

Comment:
copy-pasting the logic does not provide any guarantee of correctness. What the test is verifying right now is that you were able to do a copy-paste without mistake. But it makes it harder to change the logic, as we would need to redo the copy-paste.

I would instead change the assertion, to assert only that it is one of the allowed values (without caring about which one to avoid having to update tests when reaching the eol date indeed)

Reply:
Thanks for the feedback @stof . I will change it now.

Just to educate myself, how does this compare to the check for `4 === Kernel::MINOR_VERSION` that was currently in place for the value `symfony_lts`? Isn't it the same type of procedure?

Reply:
well, the `4 === Kernel::MINOR_VERSION` condition does not have lots of copy-pasted logic in it, and is unlikely to ever be modified (at least until we change the release schedule entirely). But it is true that it does not really guarantee the correctness either.

Reply:
Thanks for clarifying - Changes have been submitted